bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,56 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn bn_mul(bn: U256, factor: u64) -> (b256, u64) {
+    disable_panic_on_overflow();
+    let product_and_overflow = asm(bn: __addr_of(bn), factor: factor, carry, new_carry, value, current_product, product_with_carry, prod_and_ovf) {
+        // Run multiplication on the lower 64bit word
+        lw value bn i3; // load the word in (bn + 3 words) into v
```

Comment:
You could just load directly into `current_product` instead of needing to use up another register (`value`)

Reply:
I asked about this on Slack:
>Best practice here is to not re-use registers. I don’t expect any performance differences. The register allocator shouldn’t care.

Reply:
With this in mind, I think there's no good reason to reuse registers in asm blocks.